b2/account_info/in_memory.py

```diff
@@ -29,32 +30,54 @@ def inner(*args, **kwargs):
 
 class InMemoryAccountInfo(UrlPoolAccountInfo):
     def __init__(self, *args, **kwargs):
-        self.clear()
         super(InMemoryAccountInfo, self).__init__(*args, **kwargs)
+        self._clear_in_memory_account_fields()
 
     def clear(self):
-        self.set_auth_data(None, None, None, None, None, None, None)
-        self._buckets = {}
+        self._clear_in_memory_account_fields()
         return super(InMemoryAccountInfo, self).clear()
 
+    def _clear_in_memory_account_fields(self):
+        self._account_id = None
+        self._allowed = None
+        self._api_url = None
+        self._application_key = None
+        self._auth_token = None
+        self._buckets = {}
+        self._download_url = None
+        self._minimum_part_size = None
+        self._realm = None
+
     def set_auth_data(
-        self, account_id, auth_token, api_url, download_url, minimum_part_size, application_key,
-        realm
+        self,
+        account_id,
+        auth_token,
+        api_url,
+        download_url,
+        minimum_part_size,
+        application_key,
+        realm,
+        allowed=None,
     ):
+        assert self.allowed_is_valid(allowed)
         self._account_id = account_id
         self._auth_token = auth_token
         self._api_url = api_url
         self._download_url = download_url
         self._minimum_part_size = minimum_part_size
         self._application_key = application_key
         self._realm = realm
+        self._allowed = allowed
 
     def refresh_entire_bucket_name_cache(self, name_id_iterable):
         self._buckets = dict(name_id_iterable)
 
     def get_bucket_id_or_none_from_bucket_name(self, bucket_name):
         return self._buckets.get(bucket_name)
 
+    def get_bucket_name_from_allowed_or_none(self):
```

Comment:
it doesn't look like this function can return `None`. At least comment it maybe?

Reply:
Better yet, remove the function.  :-)